src/eip1559.ts

```diff
@@ -125,3 +101,30 @@ export const estimateFees = async (provider: ProviderLike): Promise<EstimationRe
     return FALLBACK_ESTIMATE;
   }
 };
+
+export const estimateFees = async (provider: ProviderLike): Promise<EstimationResult> => {
+  try {
+    const latestBlock = await getLatestBlock(provider);
+
+    if (!latestBlock.baseFeePerGas) {
+      throw new Error('An error occurred while fetching current base fee, falling back');
+    }
+
+    const baseFee = BigInt(latestBlock.baseFeePerGas);
+
+    const blockNumber = BigInt(latestBlock.number);
+
+    const feeHistory =
+      baseFee >= PRIORITY_FEE_ESTIMATION_TRIGGER
+        ? await getFeeHistory(provider, hexlify(FEE_HISTORY_BLOCKS), hexlify(blockNumber), [
+            FEE_HISTORY_PERCENTILE
+          ])
+        : undefined;
+
+    return calculateFees(baseFee, feeHistory);
+  } catch (err) {
+    // eslint-disable-next-line no-console
+    console.error(err);
```

Comment:
I know this is not fully related to this PR, but logging errors from within a library seems kind of like a bad practice to me

Reply:
Yeah, maybe for next version we figure out a better solution?